forms-flow-api/src/formsflow_api/models/process.py

```diff
@@ -45,11 +45,11 @@ class Process(AuditDateTimeMixin, AuditUserMixin, BaseModel, db.Model):
         nullable=False,
         default=ProcessStatus.DRAFT,
     )
-    tenant = db.Column(db.String(100), nullable=True)
+    tenant = db.Column(db.String(100), nullable=True, index=True)
     major_version = db.Column(db.Integer, nullable=False, index=True)
     minor_version = db.Column(db.Integer, nullable=False, index=True)
     process_key = db.Column(db.String)
-    parent_process_key = db.Column(db.String)
+    parent_process_key = db.Column(db.String, index=True)
     is_subflow = db.Column(db.Boolean, default=False)

```

Comment:
in our application we have two types of queries in this table
1. list all subflows based on tenant key
2. fetch a process by process key and checking tenant key 

so my suggestions 
```
__table_args__ = (
        # Query 1: Efficient filtering of subflows by tenant
        Index('idx_tenant_is_subflow', 'tenant', 'is_subflow'),

        # Query 2: Efficient lookup of process by tenant and process_key
        Index('idx_tenant_process_key', 'tenant', 'process_key'),
    )
```

Reply:
@auslin-aot  @sumesh-aot